src/components/Articles/LikeArticle/LikeArticle.js

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faThumbsUp, faThumbsDown } from '@fortawesome/free-solid-svg-icons';
+import PropTypes from 'prop-types';
+import { getArticleLikes, likeArticle, dislikeArticle } from '../../../actions';
+import { Button } from '../../common';
+import { clearHighlightArticleStore } from '../../../actions/articles';
+import './LikeArticle.scss';
+
+export class LikeArticle extends Component {
+  state = {
+    likeClicked: false,
+    dislikeClicked: false,
+    likesNumber: 0,
+    dislikesNumber: 0,
+    errors: {},
+    message: ''
+  };
+
+  articleLikes = (slug) => {
+    const { getArticleLikes } = this.props;
+    getArticleLikes(slug);
+  };
+
+  handleLikeClick = (e) => {
+    e.preventDefault();
+    this.setState({ likeClicked: true });
+    const {
+      article: { slug },
+      likeArticle
+    } = this.props;
+    likeArticle({ slug, status: 'like' });
+  };
+
+  handleDislikeClick = (e) => {
+    e.preventDefault();
+    this.setState({ dislikeClicked: true });
+    const {
+      article: { slug },
+      dislikeArticle
+    } = this.props;
+    dislikeArticle({ slug, status: 'dislike' });
+  };
+
+  componentWillReceiveProps(nextProps) {
+    const {
+      article: {
+        likes: { number: likesNumber },
+        dislikes: { number: dislikesNumber }
+      }
+    } = nextProps;
+
+    this.setState(prevState => ({
+      ...prevState,
+      likesNumber,
+      dislikesNumber,
+      errors: nextProps.likeErrors,
+      message: nextProps.likeMessage
+    }));
+  }
+
+  componentDidMount() {
+    const { article: { slug } } = this.props;
+    this.articleLikes(slug);
+    clearHighlightArticleStore();
+  }
+
+  render() {
+    const { likesNumber, dislikesNumber, likeClicked, dislikeClicked } = this.state;
+    const {
+      isAuth,
+      article: { slug }
+    } = this.props;
+    return (
+      <div className="row">
+        <Button
```

Comment:
could we abstract this since the two buttons are almost similar

Reply:
I tried to abstract the bottons

Reply:
how have you abstracted this?